### PR TITLE
all: remove unused code

### PR DESCRIPTION
**Pull request by petermattis**
Found using honnef.co/go/unused/cmd/unused.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5256)

### CODE REVIEW

**Comment by petermattis** [comment]
Looks like most of this is detritus left over from refactorings.

**Comment by tschottdorf** [comment]
How satisfying, LGTM. Sure hope that tool has no bugs.

**Comment by petermattis** [comment]
I swear all the tests and acceptance passed locally. Let's try again on CircleCI.

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 20 of 23 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[GLOCKFILE, line 74 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2NyoB1ptbHut2wDS:-KCv2NyoB1ptbHut2wDT:-1386208305) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/GLOCKFILE#L74)):*
any specific reason to update these dependencies but not others?

---

*[Makefile, line 238 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2Sn9YpUuMVOnVtlP:-KCv2Sn9YpUuMVOnVtlQ:204021616) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/Makefile#L238)):*
is there an upstream reference to this?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-:-KCv2maiqOgo9jui2e76:1417088721)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[Makefile, line 238 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2Sn9YpUuMVOnVtlP:-KCv2vUd90P-vORpEXRV:172265890) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/Makefile#L238)):*
https://github.com/dominikh/go-unused/issues/18?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[GLOCKFILE, line 74 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2NyoB1ptbHut2wDS:-KCv3OAPg4gdNPJXS7zj:-885910131) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/GLOCKFILE#L74)):*
No reason. This is just taking whatever `build/builder.sh glock save` gave me. 

---

*[Makefile, line 238 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2Sn9YpUuMVOnVtlP:-KCv3V7pgdHpXS1ly281:-1060016949) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/Makefile#L238)):*
I'll add that.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[GLOCKFILE, line 74 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256#-KCv2NyoB1ptbHut2wDS:-KCv7DQh52SgRXoKS_p3:1696450799) ([raw file](https://github.com/cockroachdb/cockroach/blob/4cf6935279a090b923dfeb6c96f820c154d88c55/GLOCKFILE#L74)):*
Hm, kinda sketchy but OK.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5256)*

File: Makefile [hunk 1]
```diff
@@ -233,6 +233,12 @@ check:
 	@echo "goimports"
 	@! goimports -l . | grep -vF 'No Exceptions'
 
+.PHONY: unused
+unused:
+# The -fields option generates false-positives for embedded mutex and
+# related types. See https://github.com/dominikh/go-unused/issues/18.
```
Comment (dominikh) [inline Makefile]: Fyi, that issue is closed now (and `-fields` is on by default). Hopefully there won't be any noteworthy false positives anymore.

Comment (petermattis) [inline Makefile]: Thanks for the note. Looks like the Mutex false-positives are all gone. I did find a new false-positive: https://github.com/cockroachdb/cockroach/blob/master/storage/engine/rocksdb/rocksdb.go#L38. That method (`rocksDBLog`) is exported to C so it appears unused by Go code.

Comment (dominikh) [inline Makefile]: I'll file an issue for that and see if it's possible to extract the "export" comment, thanks.

Comment (dominikh) [inline Makefile]: @petermattis That issue should be fixed now.

**Pull request merged**